benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -295,43 +304,145 @@ def update_ci_status(self, status: str, ci_minutes_time: int, benchmark_operator
+
+    def __make_run_artifacts_tarfile(self, workload: str):
+        """
+        This method tar.gz log path and return the tar.gz path
+        :return:
+        """
+        run_artifacts_path = self.__environment_variables_dict.get('run_artifacts_path', '')
+        tar_run_artifacts_path = f"{run_artifacts_path}-{self.__time_stamp_format}.tar.gz"
+        with tarfile.open(tar_run_artifacts_path, mode='w:gz') as archive:
+            archive.add(run_artifacts_path, arcname=f'{workload}-{self.__time_stamp_format}', recursive=True)
+        return tar_run_artifacts_path
+
+    @logger_time_stamp
+    def upload_run_artifacts_to_s3(self, workload: str):
+        """
+        This method uploads log to s3
+        :param workload:
+        :return:
+        """
+        s3operations = S3Operations()
+        # change workload to key convention
+        workload = workload.replace('_', '-')
+        tar_run_artifacts_path = self.__make_run_artifacts_tarfile(workload)
+        run_artifacts_hierarchy = self.__get_run_artifacts_hierarchy()
+        upload_file = f"{workload}-{self.__time_stamp_format}.tar.gz"
+        s3operations.upload_file(file_name_path=tar_run_artifacts_path,
+                                 bucket=self.__environment_variables_dict.get('bucket', ''),
+                                 key=run_artifacts_hierarchy,
+                                 upload_file=upload_file)
+        # remove local run artifacts workload folder
+        path = self.__environment_variables_dict.get('run_artifacts_path', '')
+        # verify that its not empty path
+        if len(path) > 3 and path != '/' and path and tar_run_artifacts_path and os.path.isfile(tar_run_artifacts_path):
+            # remove run_artifacts_path
+            shutil.rmtree(path=self.__environment_variables_dict.get('run_artifacts_path', ''))
+            # remove tar.gz file
+            os.remove(path=tar_run_artifacts_path)
+
+
+
 #***********************************************************************************************
 ######################################## Workloads #############################################
 #***********************************************************************************************
 
+
     @typechecked
     @logger_time_stamp
-    def stressng_pod(self, name: str=''):
+    def stressng_pod(self, name: str = ''):
         """
         This method run stressng workload
         :return:
         """
         try:
-            
             if name == '':
                 name = self.stressng_pod.__name__
             workload = name.replace('_', '-')
             kind = 'pod'
             if '_kata' in name:
                 kind = 'kata'
+            if self.__run_type == 'test_ci':
+                es_index = 'stressng-test-ci-results'
+            else:
+                es_index = 'stressng-results'
```

Comment:
Again, why is this moved outside of the `if self.__es_host` block?

Reply:
Solved empty index issue in case of Exception.
In case of failure while running pod/vm, we didnt reach to self.__es_host and move direct to Exception for upload failure date into ES. in that case we got empty index. That the reason I moved it to the beginning.